azurerm/resource_arm_app_configuration.go

```diff
@@ -0,0 +1,320 @@
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] App Configuration %q was not found in Resource Group %q - removing from state!", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on App Configuration %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if endpoint := resp.Endpoint; endpoint != nil {
+		d.Set("endpoint", resp.Endpoint)
+	}
+
+	resultPage, err := client.ListKeys(ctx, resourceGroup, name, "")
+	if err != nil {
+		return fmt.Errorf("Failed to receive access keys for App Configuration %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	values := resultPage.Values()
+	for _, value := range values {
+		accessKey := makeAccessKeyMap(value)
+
+		if strings.HasPrefix(*value.Name, "Primary") {
+			if *value.ReadOnly {
+				d.Set("primary_read_key", accessKey)
+			} else {
+				d.Set("primary_write_key", accessKey)
+			}
+		} else if strings.HasPrefix(*value.Name, "Secondary") {
+			if *value.ReadOnly {
+				d.Set("secondary_read_key", accessKey)
+			} else {
+				d.Set("secondary_write_key", accessKey)
+			}
+		} else {
+			log.Printf("[WARN] Received unknown App Configuration access key '%s', ignoring...", *value.Name)
+		}
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmAppConfigurationDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).AppConfiguration.AppConfigurationsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["configurationStores"]
+
+	fut, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(fut.Response()) {
+			return nil
+		}
+		return fmt.Errorf("Error deleting App Configuration %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := fut.Result(*client)
+	if err != nil {
+		if !response.WasNotFound(resp.Response) {
+			return nil
+		}
+		return fmt.Errorf("Error deleting App Configuration %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func validateAppConfigurationName(v interface{}, k string) (warnings []string, errors []error) {
+	value := v.(string)
+	if matched := regexp.MustCompile(`^[a-zA-Z0-9-]{5,50}$`).Match([]byte(value)); !matched {
+		errors = append(errors, fmt.Errorf("%q may only contain alphanumeric characters and dashes and must be between 5-50 chars", k))
+	}
+
+	return warnings, errors
+}
+
+func makeAccessKeyMap(value appconf.APIKey) map[string]string {
+	m := make(map[string]string)
+
+	m["id"] = *value.ID
```

Comment:
We need to check that these values aren't nil before setting them, otherwise we could panic